src/Symfony/Component/Cache/Adapter/AbstractAdapter.php

```diff
@@ -55,19 +55,21 @@ function ($key, $value, $isHit) use ($defaultLifetime) {
             null,
             CacheItem::class
         );
+        $getId = function ($key) { return $this->getId($key); };
         $this->mergeByLifetime = \Closure::bind(
-            function ($deferred, $namespace, &$expiredIds) {
+            function ($deferred, $namespace, &$expiredIds) use ($getId) {
                 $byLifetime = array();
                 $now = time();
                 $expiredIds = array();
 
                 foreach ($deferred as $key => $item) {
+                    $id = $getId(is_int($key) ? (string) $key : $key);
```

Comment:
I would call this inline (no need for the $id intermediary variable)
and I would move the cast to string inside the closure.
also, now need of the is_int check, we should unconditionally cast to string instead